src/components/FormMessage.tsx

```diff
@@ -0,0 +1,167 @@
+import Button from '@sendbird/uikit-react/ui/Button';
+import Label, {
+  LabelTypography,
+  LabelColors,
+} from '@sendbird/uikit-react/ui/Label';
+import { useCallback, useState, useEffect } from 'react';
+// eslint-disable-next-line import/no-unresolved
+import { EveryMessage } from 'SendbirdUIKitGlobal';
+import styled from 'styled-components';
+
+import Input from './FormInput';
+
+const Root = styled.div`
+  max-width: 244px;
+  display: flex;
+  flex-direction: column;
+  align-items: flex-start;
+  padding: 16px 12px;
+  gap: 8px;
+  border-radius: 16px;
+  background-color: #eeeeee;
+`;
+
+const SubmitButton = styled(Button)`
+  width: 100%;
+`;
+
+interface Field {
+  key: string;
+  title: string;
+  placeholder: string;
+  required: boolean;
+  regex: RegExp;
+  input_type: string;
+}
+interface Props {
+  message: EveryMessage;
+  form: {
+    key: string;
+    fields: Field[];
+    /** ubmitted data */
+    data: Record<string, string>;
+  };
+}
+
+type FormValues = Record<
+  string,
+  { value: string; required: boolean; hasError: boolean; isValid: boolean }
+>;
+
+export default function FormMessage(props: Props) {
+  const {
+    message,
+    form: { fields, key: formKey, data: submittedData },
+  } = props;
+  const [formValues, setInputValue] = useState<FormValues>(() =>
+    fields.reduce(
+      (acc, { key, required }) => ({
+        ...acc,
+        [key]: { value: '', required, hasError: false, isValid: false },
+      }),
+      {}
```

Comment:
Honestly, I would go against using reduce here for two reasons:
1. Hard to read.
2. Waste operation of copying accumulated into new object.

Instead I would consider using simpler `Object.entries()` interation to push key and val.

Reply:
https://github.com/sendbird/chat-ai-widget/pull/60/commits/7ec9f8499d5202b90a20dbca835adb83941afc1c is this what you mean?